directories_task.py

```diff
@@ -3,169 +3,264 @@
+            # Debug: Check current working directory
+            pwd_result = await safe_exec('pwd')
+            print(f"[Sample {sample_id}] Current working directory: {pwd_result.stdout.strip()}")
+            
+            # Get directory structure
+            output = await safe_exec('ls -R')
+            print(f"[Sample {sample_id}] Directory structure:\n{output.stdout.strip()}")
+            
+            # 1. Check depth
+            depth_cmd = "find . -type d -not -path '*/\\.*' | sort | awk -F/ '{print NF-1}' | sort -n | tail -1"
+            depth_result = await safe_exec(depth_cmd)
+            
+            if not depth_result.stdout.strip():
+                actual_depth = 0
+            else:
+                actual_depth = int(depth_result.stdout.strip())
+                
+            print(f"[Sample {sample_id}] The max depth: {actual_depth}")
+            
+            # Check depth criterion
+            expected_depth = n  # No offset needed
+            if actual_depth == expected_depth:
+                total_score += 1
+                print(f"[Sample {sample_id}] ✓ Depth criterion met (depth={actual_depth})")
+            else:
+                print(f"[Sample {sample_id}] ✗ Depth criterion not met (expected {expected_depth}, got {actual_depth})")
+            
+            # Only award breadth and uniqueness points if at least some directories were created
+            if actual_depth == 0:
+                breadth_correct = False
+                print(f"[Sample {sample_id}] ✗ Breadth criterion automatically failed (no directories created)")
+                # Set uniqueness to False as well, since no directories were created
+                dup_count = 0  # Still set to 0 for metrics tracking
+                uniqueness_passed = False
+                print(f"[Sample {sample_id}] ✗ Uniqueness criterion automatically failed (no directories created)")
+            else:
+                # 2. Check breadth - simplified approach to avoid hanging
+                breadth_correct = True
+                
+                # Only check if depth is at least 1
+                if actual_depth >= 1:
+                    # Get top-level directories
+                    top_dirs = await safe_exec("find . -mindepth 1 -maxdepth 1 -type d | sort")
+                    top_dir_list = [d for d in top_dirs.stdout.strip().split('\n') if d]
+                    
+                    # Check if we have the right number of top directories for n > 1
+                    if n > 1 and len(top_dir_list) != n:
+                        breadth_correct = False
+                        print(f"[Sample {sample_id}] ✗ Found {len(top_dir_list)} top-level directories (expected {n})")
+                    
+                    # Sample check: examine just the first directory at each level
+                    if breadth_correct and n > 1:
+                        for level in range(1, min(n, 5)):  # Limit to first 3 levels to avoid hanging
```

Comment:
Yeah I think it's a mistake to keep calling `sandbox().exec` over and over - I think that this spins up a docker container each time which is pretty expensive. Instead, can you either do most of this sandbox-stuff in a single `sandbox.exec` call, or use a single `sandbox.exec` call to extract all the info you need to do the scoring in Pythong?